code/_onclick/item_attack.dm

```diff
@@ -16,10 +16,10 @@
 
 // No comment
 /atom/proc/attackby(obj/item/W, mob/user, params)
-	return
+	return SendSignal(COMSIG_PARENT_ATTACKBY, W, user, params)
 
 /obj/attackby(obj/item/I, mob/living/user, params)
-	return I.attack_obj(src, user)
+	return ..() || I.attack_obj(src, user)

```

Comment:
does this work

Reply:
A component can return TRUE from it's signal processor to stop attack animations. Yes